test/JDBC/expected/db_collation/test_constraint_like-vu-prepare.out

```diff
@@ -0,0 +1,388 @@
+GO
+
+-- CASE 10: ESCAPE WITH LIKE
+CREATE TABLE BABEL_5608_vu_prepare_t10(BABEL_5608_vu_prepare_t10_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t10_c1 CHECK (BABEL_5608_vu_prepare_t10_col1 COLLATE Latin1_General_CI_AS LIKE '15/% %' ESCAPE '/'));
+GO
+
+CREATE VIEW BABEL_5608_vu_prepare_t10_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t10;
+GO
+
+-- CASE 11: T_CoerceViaIO
+CREATE TABLE BABEL_5608_vu_prepare_t11(BABEL_5608_vu_prepare_t11_col1 varchar(max), CONSTRAINT BABEL_5608_vu_prepare_t11_c1 CHECK (N'123' collate Latin1_General_CI_AI LIKE CAST(123 as varchar(3))));
+GO
+
+CREATE VIEW BABEL_5608_vu_prepare_t11_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t11;
+GO
+
+-- CASE 12: CFL CONDITIONS
+-- IF ELSE
+CREATE TABLE BABEL_5608_vu_prepare_t12_1 (status varchar(10), CONSTRAINT chk_status CHECK (IIF(status LIKE 'act%', 1, 0) = 1 OR IIF(status = 'inactive', 1, 0) = 1));
+GO
+
+CREATE VIEW BABEL_5608_vu_prepare_t12_1_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_1;
+GO
+
+-- CASE WHEN
+CREATE TABLE BABEL_5608_vu_prepare_t12_2 (status varchar(10),CONSTRAINT chk_status CHECK (
+        CASE 
+            WHEN status LIKE 'act%' THEN 1
+            WHEN status LIKE 'inactive' THEN 1
+            ELSE 0
+        END = 1
+    )
+);
+GO
+
+CREATE VIEW BABEL_5608_vu_prepare_t12_2_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2;
+GO
+
+-- CASE 13: VIEW WITH LIKE
+CREATE VIEW BABEL_5608_13_1_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2 WHERE status LIKE 'ac%';
+GO
+
+CREATE VIEW BABEL_5608_13_2_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2 WHERE status COLLATE estonian_ci_ai LIKE 'ac%';
+GO
+
+CREATE VIEW BABEL_5608_13_3_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2 WHERE status LIKE 'ac%' COLLATE estonian_ci_ai;
+GO
+
+CREATE VIEW BABEL_5608_13_4_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2 WHERE UPPER(status) LIKE 'ac%';
+GO
+
+CREATE VIEW BABEL_5608_13_5_VIEW AS SELECT * FROM BABEL_5608_vu_prepare_t12_2 WHERE UPPER(status) COLLATE DATABASE_DEFAULT LIKE 'ac%';
+GO
+
+-- psql
+-- CASE 1: T_Const LIKE T_CollateExpr(T_Const)
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid = 'master_dbo.BABEL_5608_vu_prepare_t1'::regclass;
+GO
+~~START~~
+text
+CHECK ((('abc'::text COLLATE sys.bbf_unicode_cp1_cs_as) ~~* (('A%'::text COLLATE sys.bbf_unicode_cp1_ci_as) COLLATE sys.bbf_unicode_cp1_cs_as)))
+~~END~~
+
+
+-- CASE 2: T_CollateExpr(T_Const) LIKE T_Const
+SELECT pg_get_constraintdef(oid) FROM pg_constraint WHERE conrelid = 'master_dbo.BABEL_5608_vu_prepare_t2'::regclass;
+GO
+~~START~~
+text
+CHECK ((((('abc'::text COLLATE sys.bbf_unicode_cp1_ci_as) COLLATE sys.bbf_unicode_cp1_cs_as) ~~* ('A%'::text COLLATE sys.bbf_unicode_cp1_cs_as)) AND ((('abc'::text COLLATE sys.bbf_unicode_cp1_ci_as) >= ('A'::text COLLATE sys.bbf_unicode_cp1_ci_as)) AND (('abc'::text COLLATE sys.bbf_unicode_cp1_ci_as) < (('A'::text COLLATE sys.bbf_unicode_cp1_ci_as) || ('￿'::text COLLATE sys.bbf_unicode_cp1_ci_as))))))
```

Comment:
why are we apply transformation rule when both sides are Const nodes? it doesnt make sense

Reply:
The constraint is this
```
'abc' COLLATE Latin1_general_ci_as LIKE 'A%'
```

So its CollateExpr LIKE Const, hence the transformation is coming into picture